src/node/gov/handlers/acks.h

```diff
@@ -0,0 +1,322 @@
+              fmt::format("No ACK record exists for member {}.", member_id));
+            return;
+          }
+
+          auto response_body = nlohmann::json::object();
+          response_body["memberId"] = member_id_str;
+          response_body["stateDigest"] = ack->state_digest;
+          ctx.rpc_ctx->set_response_json(response_body, HTTP_STATUS_OK);
+          return;
+        }
+      }
+    };
+    registry
+      .make_read_only_endpoint(
+        "/members/state-digests/{memberId}",
+        HTTP_GET,
+        api_version_adapter(get_state_digest),
+        no_auth_required)
+      .set_openapi_hidden(true)
+      .install();
+
+    auto update_state_digest = [&](auto& ctx, ApiVersion api_version) {
+      switch (api_version)
+      {
+        case ApiVersion::preview_v1:
+        default:
+        {
+          // Get memberId from path parameter
+          std::string error;
+          std::string member_id_str;
+          if (!ccf::endpoints::get_path_param(
+                ctx.rpc_ctx->get_request_path_params(),
+                "memberId",
+                member_id_str,
+                error))
+          {
+            detail::set_gov_error(
+              ctx.rpc_ctx,
+              HTTP_STATUS_BAD_REQUEST,
+              ccf::errors::InvalidResourceName,
+              std::move(error));
+            return;
+          }
+
+          // Confirm this matches memberId from signature
+          ccf::MemberId member_id(member_id_str);
+          const auto& cose_ident =
+            ctx.template get_caller<ccf::MemberCOSESign1AuthnIdentity>();
+          if (cose_ident.member_id != member_id)
+          {
+            detail::set_gov_error(
+              ctx.rpc_ctx,
+              HTTP_STATUS_BAD_REQUEST,
+              ccf::errors::InvalidAuthenticationInfo,
+              fmt::format(
+                "Member ID from path parameter ({}) does not match "
+                "member ID from body signature ({}).",
+                member_id,
+                cose_ident.member_id));
+            return;
+          }
+
+          ccf::MemberAck ack;
+
+          // Get previous ack, if it exists
+          auto acks_handle =
+            ctx.tx.template rw<ccf::MemberAcks>(Tables::MEMBER_ACKS);
+          auto ack_opt = acks_handle->get(member_id);
+          if (ack_opt.has_value())
+          {
+            ack = ack_opt.value();
+          }
+
+          // Get signature, containing merkle root state digest
+          auto sigs_handle =
+            ctx.tx.template ro<ccf::Signatures>(Tables::SIGNATURES);
+          auto sig = sigs_handle->get();
+          if (!sig.has_value())
+          {
+            // TODO: Behaviour change, why wasn't this an error previously?
+            detail::set_gov_error(
+              ctx.rpc_ctx,
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::InternalError,
+              "Service has no signatures to ack yet - try again soon.");
+            return;
+          }
```

Comment:
Behaviour change compared to the `update_state_digest` endpoint in `member_frontend.h`:

```
        auto s = sig->get();
        if (s)
        {
          ma->state_digest = s->root.hex_str();
          mas->put(member_id.value(), ma.value());
        }
```

If there's no signatures yet, we say the member can ack an empty (null?) state digest. But we don't even write that back to the KV. I think that's unsafe? This can only happen right at the start of a service, before a signature has been emitted, so will likely never occur in practice, but I err towards a clear error if it does occur.